### PR TITLE
Refactor Texture creation

### PR DESCRIPTION
**Pull request by lilleyse**
For #640

### CODE REVIEW

File: Source/Renderer/CubeMap.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ define([
         '../Core/destroyObject',
         '../Core/DeveloperError',
         '../Core/Math',
+        '../Core/PixelFormat',
```
Comment (pjcozzi) [inline Source/Renderer/CubeMap.js]: This is no longer needed in `Context`.

File: Source/Renderer/CubeMap.js [hunk 2]
```diff
@@ -28,11 +30,147 @@ define([
...
+     * @exception {DeveloperError} Each face in options.sources must have the same width and height.
+     * @exception {DeveloperError} options requires a source field to create an initialized cube map or width and height fields to create a blank cube map.
+     * @exception {DeveloperError} Width must equal height.
+     * @exception {DeveloperError} Width and height must be greater than zero.
+     * @exception {DeveloperError} Width and height must be less than or equal to the maximum cube map size.
+     * @exception {DeveloperError} Invalid options.pixelFormat.
+     * @exception {DeveloperError} options.pixelFormat cannot be DEPTH_COMPONENT or DEPTH_STENCIL.
+     * @exception {DeveloperError} Invalid options.pixelDatatype.
+     *
+     * @see Context#createSampler
```
Comment (pjcozzi) [inline Source/Renderer/CubeMap.js]: Remove this.

File: Source/Renderer/CubeMap.js [hunk 3]
```diff
@@ -28,11 +30,147 @@ define([
...
+     *
      * @private
      */
-    var CubeMap = function(gl, textureFilterAnisotropic, textureTarget, texture, pixelFormat, pixelDatatype, size, preMultiplyAlpha, flipY) {
+
```
Comment (pjcozzi) [inline Source/Renderer/CubeMap.js]: Remove whitespace

File: Source/Renderer/CubeMap.js [hunk 4]
```diff
@@ -28,11 +30,147 @@ define([
...
+    var CubeMap = function(options) {
+
```
Comment (pjcozzi) [inline Source/Renderer/CubeMap.js]: Remove whitespace

File: Source/Renderer/CubeMap.js [hunk 5]
```diff
@@ -28,11 +30,147 @@ define([
...
+        if (PixelFormat.isDepthFormat(pixelFormat)) {
+            throw new DeveloperError('options.pixelFormat cannot be DEPTH_COMPONENT or DEPTH_STENCIL.');
+        }
+
+        if (!PixelDatatype.validate(pixelDatatype)) {
+            throw new DeveloperError('Invalid options.pixelDatatype.');
+        }
+        //>>includeEnd('debug');
+
+        if ((pixelDatatype === PixelDatatype.FLOAT) && !context.floatingPointTexture) {
```
Comment (pjcozzi) [inline Source/Renderer/CubeMap.js]: I think it is OK for `//>>includeEnd('debug');` to come after this.

File: Source/Renderer/Texture.js [hunk 6]
```diff
@@ -152,6 +177,104 @@ define([
...
+     * @returns {Texture} A texture with contents from the framebuffer.
+     *
+     * @exception {DeveloperError} Invalid pixelFormat.
+     * @exception {DeveloperError} pixelFormat cannot be DEPTH_COMPONENT or DEPTH_STENCIL.
+     * @exception {DeveloperError} framebufferXOffset must be greater than or equal to zero.
+     * @exception {DeveloperError} framebufferYOffset must be greater than or equal to zero.
+     * @exception {DeveloperError} framebufferXOffset + width must be less than or equal to canvas.clientWidth.
+     * @exception {DeveloperError} framebufferYOffset + height must be less than or equal to canvas.clientHeight.
+     *
+     * @see Context#createSampler
```
Comment (pjcozzi) [inline Source/Renderer/Texture.js]: Remove this.

**Comment by pjcozzi** [comment]
Looks good, just those minor comments.  Let me know when this is ready.

**Comment by lilleyse** [comment]
Added the fixes

**Comment by pjcozzi** [comment]
Looks good, thanks.

**Pull request merged**